src/simulations/modeling/driven.py

```diff
@@ -165,3 +160,144 @@ def adjust_fluxes2model(
+    proteomics: list(dict)
+        List of measurements matching the `Proteomics` schema.
+    growth_rate: dict
+        Growth rate, matching the `GrowthRate` schema.
+
+    Returns
+    -------
+    growth_rate: dict
+        New growth rate (will change if the model couldn't grow at the inputted value).
+    proteomics: list(dict)
+        Filtered list of proteomics.
+    """
+
+    # reset growth rate in model:
+    model.reactions.get_by_id(biomass_reaction).bounds = (0, 1000)
+
+    # build a table with protein ids, met ids in model and values to constrain with:
+    prot_df = pd.DataFrame()
+    for protein in proteomics:
+        protein_id = protein["identifier"]
+        lb, ub = bounds(protein["measurement"], protein["uncertainty"])
+        for met in model.metabolites:
+            if protein_id in met.id:
+                new_row = pd.DataFrame(
+                    data={"met_id": met.id, "value": ub}, index=[protein_id]
+                )
+                prot_df = prot_df.append(new_row)
+
+    # constrain the model with all proteins and optimize:
+    limit_proteins(model, prot_df["value"])
+    solution = model.optimize()
+    new_growth_rate = solution.objective_value
+
+    # while the model cannot grow to the desired level, remove the protein with
+    # the highest shadow price:
+    minimal_growth, ub = bounds(growth_rate["measurement"], growth_rate["uncertainty"])
+    prots_to_remove = []
+    while new_growth_rate < minimal_growth and not prot_df.empty:
+        # get most influential protein in model:
+        top_protein = top_shadow_prices(solution, list(prot_df["met_id"]))
+        top_protein = top_protein.index[0]
+        top_protein = prot_df.index[prot_df["met_id"] == top_protein][0]
+
+        # update data: append protein to list, remove from current dataframe and
+        # increase the corresponding upper bound to +1000:
+        prots_to_remove.append(top_protein)
+        prot_df = prot_df.drop(labels=top_protein)
+        limit_proteins(model, pd.Series(data=[1000], index=[top_protein]))
+
+        # re-compute solution:
+        solution = model.optimize()
+        if solution.objective_value == new_growth_rate:  # the algorithm is stuck
+            break
+        new_growth_rate = solution.objective_value
+
+    # update growth rate if optimization was not successful:
+    if new_growth_rate < minimal_growth:
+        if growth_rate["uncertainty"]:
+            growth_rate["measurement"] = new_growth_rate + growth_rate["uncertainty"]
+        else:
+            growth_rate["measurement"] = new_growth_rate
+
+    # update proteomics by removing flexibilized proteins:
+    for protein in prots_to_remove:
+        index = next(
+            (
+                index
+                for (index, d) in enumerate(proteomics)
+                if d["identifier"] == protein
+            ),
+            None,
+        )
+        proteomics.pop(index)
```

Comment:
`del proteomics[index]` is a bit more explicit when you don't care about returning the item.

Reply:
done in commit [`1fe6f9a`](https://github.com/DD-DeCaF/simulations/pull/147/commits/1fe6f9ad49497691b7f7feea0d7a10233907d456)